contract/migrations/17.0.1.4.7/pre-migration.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2025 Tecnativa - Carlos Lopez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    _new_columns = [("contract.line", "partner_id", "many2one")]
+    openupgrade.add_columns(env, _new_columns)
+    openupgrade.logged_query(
```

Comment:
You don't need this, as related fields with only one dot are automatically populated by SQL in the ORM framework.